src/CodeGenerator/AutoloadGenerator.php

```diff
@@ -7,87 +7,93 @@
 
 namespace Zend\Di\CodeGenerator;
 
-use Zend\Code\Generator\DocBlockGenerator;
-use Zend\Code\Generator\FileGenerator;
-use Zend\Code\Generator\ClassGenerator;
-use Zend\Code\Generator\MethodGenerator;
-use Zend\Code\Generator\PropertyGenerator;
-use Zend\Code\Generator\PropertyValueGenerator;
+use SplFileObject;
+use Throwable;
+use Zend\Di\Exception\GenerateCodeException;
+use function array_keys;
+use function array_map;
+use function file_get_contents;
+use function implode;
+use function str_repeat;
+use function strtr;
+use function var_export;
 
 class AutoloadGenerator
 {
     use GeneratorTrait;
 
-    private $namespace;
+    const CLASS_TEMPLATE = __DIR__ . '/../../templates/autoloader-class.template';
+    const FILE_TEMPLATE = __DIR__ . '/../../templates/autoloader-file.template';
 
     /**
-     * @param string $namespace
+     * @var string
      */
+    private $namespace;
+
     public function __construct(string $namespace)
     {
         $this->namespace = $namespace;
     }
 
-    private function generateAutoloaderClass(array &$classmap)
+    private function writeFile(string $filename, string $code) : void
+    {
+        try {
+            $file = new SplFileObject($filename, 'w');
+            $file->fwrite($code);
+        } catch (Throwable $e) {
+            throw new GenerateCodeException(sprintf('Failed to write output file "%s"', $filename), 0, $e);
```

Comment:
Kept for BC. Can we drop it safely and let SplFileObject throw?

Reply:
I would argue to keep it, in case any end-users are trying to catch that specific exception type.